src/composing/platt_scaling.rs

```diff
@@ -182,12 +182,19 @@ where
             targets.len(),
             "The number of data points must match the number of output targets."
         );
-        *targets = self
+        for (x, target) in self.obj.predict(data).iter().zip(targets.iter_mut()) {
+            *target = platt_predict(*x, self.a, self.b);
+        }
+
+        // TODO: why did we return a len 98 array when data had 100 rows??
+        let before: Array1<Pr> = self
             .obj
             .predict(data)
             .iter()
             .map(|x| platt_predict(*x, self.a, self.b))
             .collect();
+
+        assert_eq!(targets, &before);
```

Comment:
Is this a bug in `self.obj.redict()`?

Reply:
It is a bug in `generate_dummy_values` I think.

Reply:
https://github.com/rust-ml/linfa/blob/aeb354891323cca79d585d67ab67582237521643/src/composing/platt_scaling.rs#L461 this has to be 102 :D

Reply:
we are not using `1.0` and `0.0` because of the inverse logit transformation

Reply:
I instead made the `linspace` make `n` values instead of `n - 2`. It is more intuitive that way.